platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/EdgeState.java

```diff
@@ -176,7 +176,7 @@ void detachFromTargetNodes() {
      * end fail resolution.
      */
     void failWith(Throwable err) {
-        targetNodeSelectionFailure = new ModuleVersionResolveException(dependencyState.getRequested(), err);
+        targetNodeSelectionFailure = new ModuleVersionResolveException(selector.getSelector(), err);
```

Comment:
What's the difference between these?

Reply:
`dependencyState.getRequested()` is pre-substitution. 
`selector.getSelector()` is post-substitution. 

The value passed in here is used when constructing a `DefaultUnresolvedDependencyResult`, where it is used as the return value for `getAttempted`. `getAttempted` represents the _actual_ component selector we tried to resolve. 

So, we should be constructing this failure with the attempted selector, not the requested selector.